tests/unit/test_replay_manager.py

```diff
@@ -1,94 +1,249 @@
     assert nclist.count() == 1
     netcon1 = nclist.o(0)
     assert netcon1.weight[4] == int(NetConType.NC_PRESYN)
 
     # get the netcon generated by ReplayStim for connection 1->1001
     nclist = Nd.cvode.netconlist("", cell, "")
+    replay_netcons = []
     for netcon in nclist:
-        if netcon.srcgid() < 0:
-            assert netcon.weight[4] == int(NetConType.NC_REPLAY)
+        if netcon.srcgid() < 0 and netcon.weight[4] == int(NetConType.NC_REPLAY):
+            replay_netcons += [netcon]
             assert netcon.weight[0] == netcon1.weight[0]
             assert netcon.delay == netcon1.delay
             assert netcon.threshold == 10
+    assert len(replay_netcons) > 0
+
+    # get voltage variations in cell 1001
+    cell = nd.circuits.get_node_manager("RingB").get_cell(1001)
+    voltage_vec = Nd.Vector()
+    voltage_vec.record(cell._cellref.soma[0](0.5)._ref_v)
+
+    Nd.finitialize()
+    nd.run()
+
+    expected_positions = np.array([39, 81])
+    utils.check_signal_peaks(voltage_vec, expected_positions)
+
+
+@pytest.mark.parametrize("create_tmp_simulation_config_file", [
+    {
+        "simconfig_fixture": "ringtest_virtual_pop_config",
+        "extra_config": {
+            "inputs": {
+                "spikeReplay" : {
+                    "module": "synapse_replay",
+                    "input_type": "spikes",
+                    "spike_file": INPUT_SPIKES_FILE,
+                    "delay": 0,
+                    "duration": 50,
+                    "node_set": "RingB",
+                }
+            }
+        }
+    },
+], indirect=True)
+@pytest.mark.forked
+def test_replay_virtual_population(create_tmp_simulation_config_file):
+    from neurodamus import Neurodamus
+    from neurodamus.core import NeurodamusCore as Nd
+    from neurodamus.core.configuration import Feature
+    from neurodamus.connection import NetConType
+
+    nd = Neurodamus(
+        create_tmp_simulation_config_file,
+        restrict_features=[Feature.Replay],
+        disable_reports=True,
+        cleanup_atexit=False,
+    )
+
+    edges_a_b = nd.circuits.get_edge_manager("RingA", "RingB")
+    conn_1_1001 = next(edges_a_b.get_connections(1001, 1))
+    time_vec = conn_1_1001._replay.time_vec.as_numpy()
+    npt.assert_allclose(time_vec, [0.1, 2.275, 4.35, 7.725])
+    # # get the netcon generated by ReplayStim for connection 1->1001
+    cell = nd._pc.gid2cell(1001)
+    nclist = Nd.cvode.netconlist("", cell, "")
+
+    replay_netcons = []
+    for netcon in nclist:
+        if netcon.srcgid() < 0 and netcon.weight[4] == int(NetConType.NC_REPLAY):
+            replay_netcons += [netcon]
+    assert len(replay_netcons) > 0
+
+    # get voltage variations in cell 1001
+    cell = nd.circuits.get_node_manager("RingB").get_cell(1001)
+    voltage_vec = Nd.Vector()
+    voltage_vec.record(cell._cellref.soma[0](0.5)._ref_v)
+
+    Nd.finitialize()
+    nd.run()
+
+    expected_positions = np.array([39, 81])
```

Comment:
are you sure you need to convert it to a `np.array` explicitly? I think it works with a simple list too

Reply:
even in that case you can handle the conversion in the function itself no?

Reply:
this is still missing no?